common/core/web/keyboard-processor/src/keyboards/defaultLayouts.ts

```diff
@@ -176,6 +176,11 @@ namespace com.keyman.keyboards {
       if(formFactor == 'desktop') {
         invalidIdList = Layouts.generateLayerIds(chiral);
 
+        // Only display these by default for the desktop layout.
+        // Mobile form-factors must have a pre-defined layout in order 
+        // to display underlying keys.
+        layout["displayUnderlying"] = !!keyboard.scriptObject['KDU'];
```

Comment:
This is a bit of an assumption on my part, which will be affected by whatever the right answer to [this comment](https://github.com/keymanapp/keyman/issues/3950#issuecomment-789358087) from the base issue is:

> Hmm... that does leave one last question. Suppose the following scenario:
> 
> A keyboard that:
> 
>     * does not specify a touch layout, relying upon KMW defaults
> 
>     * does specify the `KDU` property
> 
>     * is being used on a touch device.
> 
> 
> Since the resulting layout is "desktop-based", and there is no layout pre-defined for that form factor, should this scenario utilize the flag or not?

This particular code line is specifically for that exact scenario.

Reply:
Yes. In a scenario where there is no touch layout defined, use the `KDU` property from the desktop layout.